apps/root/src/pages/earn-access-now/components/elegibility-confirmation/use-cases.tsx

```diff
@@ -0,0 +1,242 @@
+      />
+    </StyledViewableButton>
+  </>
+);
+
+export const NotEligibleStatus = () => {
+  const openConnectWalletModal = useOpenConnectModal();
+  const trackEvent = useTrackEvent();
+
+  const onConnectWallet = React.useCallback(() => {
+    openConnectWalletModal(WalletActionType.link);
+    trackEvent('Earn Early Access - Elegibility confirmation - Link wallet');
+  }, [openConnectWalletModal, trackEvent]);
+  return (
+    <>
+      <TitleContainer>
+        <StyledTitle>
+          <FormattedMessage
+            description="earn-access-now.eligibility.confirmation.not-eligible.title"
+            defaultMessage="You're Not Eligible 😔"
+          />
+        </StyledTitle>
+        <StyledDescription>
+          <FormattedMessage
+            description="earn-access-now.eligibility.confirmation.not-eligible.description"
+            defaultMessage="It seems none of your wallets meet the criteria for Earn Early Access. Don't worry! You can still join by trying another wallet or finding a referral code on our Discord or Twitter!"
+          />
+        </StyledDescription>
+      </TitleContainer>
+      <Button variant="outlined" onClick={onConnectWallet}>
+        <FormattedMessage
+          description="earn-access-now.eligibility.confirmation.not-eligible.button"
+          defaultMessage="Try with another wallet"
+        />
+      </Button>
+    </>
+  );
+};
+
+export const ElegibleCase = () => {
+  // TODO: Specify elegible wallet address if isAuth
+
+  return (
+    <>
+      <TitleContainer>
+        <StyledTitle>
+          <FormattedMessage
+            description="earn-access-now.eligibility.confirmation.eligible.title"
+            defaultMessage="You're Eligible for Earn Early Access! 🎉"
+          />
+        </StyledTitle>
+        <StyledDescription>
+          <FormattedMessage
+            description="earn-access-now.eligibility.confirmation.eligible.description"
+            defaultMessage="Your wallet meets the eligibility criteria. <b>You now have full access to Earn with all your connected wallets</b>. Start exploring yield strategies today!"
+            values={{
+              b: (chunks: React.ReactNode) => <b>{chunks}</b>,
+            }}
+          />
+        </StyledDescription>
+      </TitleContainer>
+      <ClaimEarnButton />
+    </>
+  );
+};
+
+export const NeedsSignatureCase = ({
+  elegibleWallets,
+  setStatus,
+}: {
+  elegibleWallets: ViemAddress[];
+  setStatus: (status: ElegibilityConfirmationStatus) => void;
+}) => {
+  const wallets = useWallets();
+  const [walletToSign, setWalletToSign] = React.useState<ViemAddress | null>(null);
+
+  React.useEffect(() => {
+    if (elegibleWallets.length === 1) {
+      setWalletToSign(elegibleWallets[0]);
+    }
+  }, [elegibleWallets]);
+
+  const requestSignature = () => {
+    if (!walletToSign) return;
+    // TODO: Request signature
+    setStatus(ElegibilityConfirmationStatus.OWNERSHIP_CONFIRMED);
```

Comment:
we will need the status to connect the wallet

Reply:
Good catch sir